packages/client/tests/functional/composites-mongo-object/_matrix.ts

```diff
@@ -0,0 +1,30 @@
+import { defineMatrix } from '../_utils/defineMatrix'
+
+const testMatrix = defineMatrix(() => [
+  [
+    {
+      provider: 'mongodb',
+    },
+  ],
+  [
+    {
+      contentProperty: 'required',
+    },
+
+    {
+      contentProperty: 'optional',
```

Comment:
Very nice to see how you condensed these tests :100: Definitely what I had in mind when I wrote them 😁